common/types/timestamp.go

```diff
@@ -272,12 +274,33 @@ func timestampGetMillisecondsWithTz(t time.Time, tz ref.Val) ref.Val {
 
 func timeZone(tz ref.Val, visitor timestampVisitor) timestampVisitor {
 	return func(t time.Time) ref.Val {
+		var err error
+		var offset int
 		if StringType != tz.Type() {
 			return ValOrErr(tz, "no such overload")
 		}
-		loc, err := time.LoadLocation(string(tz.(String)))
-		if err == nil {
-			return visitor(t.In(loc))
+		val := string(tz.(String))
+		ind := strings.Index(val, ":")
+		if ind == -1 {
+			loc, err := time.LoadLocation(val)
+			if err == nil {
+				return visitor(t.In(loc))
+			}
+		} else {
+			hr, err := strconv.Atoi(string(val[0:ind]))
+			min, err2 := strconv.Atoi(string(val[ind+1]))
+			if err == nil && err2 == nil {
+				if string(val[0]) == "-" {
+					offset = hr*60 - min
+				} else {
+					offset = hr*60 + min
+				}
+				secondsEastOfUTC := int((time.Duration(offset) * time.Minute).Seconds())
+				timezone := time.FixedZone("Time Zone", secondsEastOfUTC)
+				curr_time := time.Date(t.Year(), t.Month(), t.Day(), t.Hour(), t.Minute(), t.Second(), t.Nanosecond(), timezone)
```

Comment:
The timezone is a valid location - any reason not to just use visitor(t.In(timezone))?